### ISSUE TITLE
Listen Module - Testing Mar 26th

### ISSUE THREAD

**Issue by Brahmanathaswami** [Himalayan-Academy/Siva-Siva-App#275]
1) Testing Journal:

Desktop: 
  - Downloads: there is a song on disk. It does require an HTTP call
  - We play it
  - when it comes to the four second, tap on the journal icon. it does up/down
  - Close the Listen
  - We go Home; click on the Journal, we go the Journal; we show up, songs, 4 Sec. marker. Why it shows a 1/60 of a second we can't say why it shows a timestamp, not in Min/secs...
  - Click on Journal; right away, "success"we go the song, pick up at the 4 sec timestamp
  - We allow the songs to play to a 10 second, click on  Journal
  - We go Home; click on Journal. There is a second entry. Tagged to the 10 second marker
  - Click on Journal; we go to the song at "listen-file" it says min/sec
  - it show a none response: cannot click on play. Doesn't play
  - Go Home; Journal; click an 4 Second marker, now it plays; show the timestamp marker

Journal Entry:

<img width="414" alt="image" src="https://user-images.githubusercontent.com/3675722/111800285-e3d3fb80-886f-11eb-9aaa-45431f28cafa.png">

First item in Journal success, it plays, can pause

<img width="409" alt="image" src="https://user-images.githubusercontent.com/3675722/111800781-61980700-8870-11eb-84d5-c468303620d0.png">

Seconds item in Journal, on top show a 10 sec marker
we go to "listen-file"  and get no response: can not play

<img width="414" alt="image" src="https://user-images.githubusercontent.com/3675722/111801414-0581b280-8871-11eb-8866-14cb5c9527fa.png">

**Comment by Brahmanathaswami** [comment]
And BTW http calls getting no response

<img width="414" alt="image" src="https://user-images.githubusercontent.com/3675722/111803308-0582b200-8873-11eb-859d-204168e83425.png">

**Comment by Brahmanathaswami** [comment]
We are not getting "hard crash" 

1) the download  play; it  goes at line 990 (Chunk: errpr in  object expression) near "7" char 8
2) the play button takes back to the beginning

<img width="447" alt="image" src="https://user-images.githubusercontent.com/3675722/112707771-e2976580-8e51-11eb-9178-e53f6cad4ebd.png">

**Comment by Brahmanathaswami** [comment]
on behavior_Listen 

you can (if you want to do it )

---------------------
line 990
  --put the long id of pTarget into sLastTarget 
   put pTarget into sLastTarget
 
--------------
  in group  "DownloadControls"  in your behavior_Download
  
  line 157
  # you can delete these two lines
     -- delete word 1 to 4 of sRequestedBy
      -- comment dispatch "updateCachedListings"  to sRequestedBy with sRecordNumber
  
      hidedDownloadControls
      put removeQuotes(sRequestedBy) into tModule     
      tracking_RecordActivity tModule,"download/saved"
   end if
end downloadComplete

**Comment by Brahmanathaswami** [comment]
card "listen-my-audio"  we get there from the header "My Downloads"  screen: 

card  "listen-my-audio": Songs / Audio Books / Talks: should start at zero NOT the starting timestamp of the Journal. They should only start at the current timestamps from the journal be clicking journal icon / entry in the journals

1. When go the card "listen-file" from the "My Downloads" link: timestamps should set to empty or 0
2. When coming from "Journal" the card "listen-file" should start to the timestamp which given last by the user for journal entry.

This usage was taking into effect before you started work. Now it does not work at all.
:-(

The timestamp on mobile is correct: hours | minutes | seconds. 
That is good
:-)

**Comment by Brahmanathaswami** [comment]
card "listen-my-audio":  the "My Downloads" on mobile field "AudioList" is currently higher to some number of pixel, that is not addressed in the updateUI

line 251:
    case "listen-my-audio"
    
    # this should  be there
      put the long id of fld "audioList" into tAudioList
         set the rect of tAudioList to (10, the bottom of tCollectionsLabel + 18, \
               CardWidth() - 7, the top of tFooter - 17)
         centerMe tAudioList, 60,70

**Comment by bwmilby** [comment]
CenterMe shouldn’t be needed.  If the rect isn’t correct, then just adjust it directly.  It is being sized and positioned relative to other controls.

**Comment by soapdog** [comment]
I haven't touched any geometry code or changed any of the UI. I'm only working on the behaviors and other plain-text stacks.

**Comment by soapdog** [comment]
> This usage was taking into effect before you started work. Now it does not work at all.

In issue #274, I've outlined how buggy the listen module was, be aware that that was not an exhaustive list, I've found a ton more after working that issue. 

Don't blame me about the current broken state of the module. I'm still working trying to untangle it.  I never claimed it to be working, what I've been saying is where it is breaking after some more work was done.

That module is a maze of confusing state management and functions that don't behave exactly like they're written — such as passing a _clickline_ to a variable called _pItemID_, and then trying to fetch it from the database, or impossible _if clauses_ that will never be met because the variables they're checking are never ever set those values.

As can be seen on my last commit message to the _listen-5_ branch: https://github.com/Himalayan-Academy/Siva-Siva-App/commit/9a4a97c286be0aff87b49cddd3f890e0c9ce8508

```
 [[ listen ]] hard crash on iOS Simulator 
```

The stack is broken, it is not ready for testing, it is not supposed to be working.
